### PR TITLE
Move scheduler specific rendering code to SchedulerTrack.

### PR DESCRIPTION
**Pull request by pierricgimmig**
- Simplify ThreadTrack mega rendering function
    - Add util functions to Batcher
    - Simplify EventTrack
    - Remove obsolete code in TimeGraph and TimeGraphLayout

This change cleans up the mega ThreadTrack::UpdatePrimitives functions by extracting the scheduling specific code to SchedulerTrack::UpdatePrimitives.  Some helper functions were added to make code easier to read.

### CODE REVIEW

File: OrbitGl/Batcher.cpp [hunk 1]
```diff
@@ -5,17 +5,92 @@
...
     }
   }
 
   return nullptr;
+}
+
+void Batcher::GetBoxGradientColors(Color color, Color* colors) {
+  const float kGradientCoeff = 0.94f;
+  Vec3 dark = Vec3(color[0], color[1], color[2]) * kGradientCoeff;
+  colors[0] = Color((unsigned char)dark[0], (unsigned char)dark[1],
```
Comment (dimitry-) [inline OrbitGl/Batcher.cpp:86]: static_cast<uint32_t>

File: OrbitGl/ThreadTrack.cpp [hunk 2]
```diff
@@ -58,189 +58,134 @@ std::string GetExtraInfo(const Timer& timer) {
...
+  const Color kInactiveColor(100, 100, 100, 255);
+  const Color kSelectionColor(0, 128, 255, 255);
+  if (is_selected) {
+    return kSelectionColor;
+  } else if (inactive) {
+    return kInactiveColor;
+  }
+
+  Color color = time_graph->GetTimesliceColor(timer);
+  const uint8_t kOddAlpha = 210;
```
Comment (dimitry-) [inline OrbitGl/ThreadTrack.cpp:73]: constexpr?

File: OrbitGl/ThreadTrack.cpp [hunk 3]
```diff
@@ -58,189 +58,134 @@ std::string GetExtraInfo(const Timer& timer) {
...
+  return track_y - layout.GetEventTrackHeight() -
+         layout.GetSpaceBetweenTracksAndThread() -
+         layout.GetTextBoxHeight() * (depth + 1);
+}
+
+//-----------------------------------------------------------------------------
+void ThreadTrack::SetTimesliceText(const Timer& timer, double elapsed_us,
+                                   float min_x, TextBox* text_box) {
+  if (text_box->GetText().empty()) {
+    double elapsed_millis = ((double)elapsed_us) * 0.001;
```
Comment (dimitry-) [inline OrbitGl/ThreadTrack.cpp:93]: you probably do not need this cast here - 0.001 should do the job for you

File: OrbitGl/Batcher.cpp [hunk 4]
```diff
@@ -5,17 +5,95 @@
...
+void Batcher::AddBox(const Box& a_Box, const Color* colors,
+                     PickingID::Type picking_type, void* user_data) {
+  Color pickCol = PickingID::GetColor(picking_type, box_buffer_.m_Boxes.size());
+  box_buffer_.m_Boxes.push_back(a_Box);
+  box_buffer_.m_Colors.push_back(colors, 4);
+  box_buffer_.m_PickingColors.push_back_n(pickCol, 4);
+  box_buffer_.m_UserData.push_back(user_data);
+}
+
+void Batcher::AddBox(const Box& a_Box, const Color color, PickingID::Type picking_type,
```
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply OrbitGl/Batcher.cpp]: Color is 32 bit, const doesn't make sense here you're right, we can pass it by value.

File: OrbitGl/Batcher.cpp (hunk 1 above)
Comment (pierricgimmig) [reply OrbitGl/Batcher.cpp:86]: done, static_cast<uint8_t>.

> Can you cleanup the commit history? It's six commits but it seems like it should only be one.

@dpallotti , yes I'll "squash and merge", this was indeed intended to be submitted as a single commit.

File: OrbitGl/ThreadTrack.cpp (hunk 2 above)
Comment (pierricgimmig) [reply OrbitGl/ThreadTrack.cpp]: done

File: OrbitGl/ThreadTrack.cpp (hunk 3 above)
Comment (pierricgimmig) [reply OrbitGl/ThreadTrack.cpp]: true!  done.

File: OrbitGl/Batcher.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply OrbitGl/Batcher.cpp]: Done.

**Pull request merged**